concordium_p2p_rpc.proto

```diff
@@ -4,223 +4,433 @@ import "google/protobuf/wrappers.proto";
 
 package concordium;
 
+// An empty message.
 message Empty {}
 
+// A numeric response.
 message NumberResponse {
   uint64 value = 1 [jstype = JS_STRING];
 }
 
+// A response consisting of a boolean.
 message BoolResponse {
   bool value = 1;
 }
 
+// A response in string format.
 message StringResponse {
   string value = 1;
 }
 
+// A response that is encoded in JSON.
+// JSON schemas are available at https://developer.concordium.software/en/mainnet/net/references/grpc.html.
 message JsonResponse {
   string value = 1;
 }
 
+// A response in binary format.
 message BytesResponse {
   bytes value = 1;
 }
 
+// A request that suggests the node to connect to the specified peer.
 message PeerConnectRequest {
+  // The IP of the peer.
   google.protobuf.StringValue ip = 1;
+
+  // The port of the peer.
   google.protobuf.Int32Value port = 2;
 }
 
+// A peer node.
 message PeerElement {
+  // The unique id of the node.
   google.protobuf.StringValue node_id = 1;
+
+  // The port of the node.
   google.protobuf.UInt32Value port = 2;
+
+  // The IP of the node.
   google.protobuf.StringValue ip = 3;
+
   enum CatchupStatus {
+    // The node has caught up from this peer and is, for now, up to date with the peer.
+    // If the node receives messages with novel information from a peer with this status,
+    // then the peer's status is changed to 'PENDING'.
     UPTODATE = 0;
+    // The peer is in the node's round-robin queue of peers to catch up from.
+    // New peers will start with this status.
```

Comment:
```suggestion
    // The peer might have some data unknown to us. A peer can be in this state either because
    // it sent a message that refers to data unknown to us, or before we have established a baseline with it.
    // The latter happens during node startup, as well as upon protocol updates until the initial catchup handshake
    // completes.
```